Import wget-1.21.3-3.fc37.src.rpm as part of XS-1557

glehmann opened the pull request:
* rebuild for openssl3
scratch build: https://koji.xcp-ng.org/taskinfo?taskID=89704

ydirson reviewed (changes requested):
upstream is too different from XS, and fedora f39 did have a 1.21 - the base might have been the last one before they upgraded to 1.21.4.  Depending on the actual level of changes from XS we might even want to directly use the latest upstream?

rzr commented on SPECS/wget.spec:
```diff
@@ -0,0 +1,107 @@
+# We will no longer use wget in XS9, so it's safe to skip the unit test in XS9.
+BuildRequires: perl(lib)
+BuildRequires: perl(English)
+BuildRequires: perl(HTTP::Daemon)
```
Can  xcp-ng-8 use `perl(HTTP::Daemon)`  provided by `perl-HTTP-Daemon` ? 

It failed to build:

https://koji.xcp-ng.org/kojifiles/work/tasks/9706/89706/root.log

```
DEBUG util.py:461:  Error: No Package found for perl(HTTP::Daemon)
```

I guess base package `perl-HTTP-Daemon-6.01-8.el7.noarch` should be added to build system.

```
repoquery --whatprovides  'perl(HTTP::Daemon)' --enablerepo='base'
perl-HTTP-Daemon-0:6.01-8.el7.noarch

repoquery --enablerepo='base' --location  'perl-HTTP-Daemon'

https://vault.centos.org/centos/7/os/x86_64/Packages/perl-HTTP-Daemon-6.01-8.el7.noarch.rpm

```

From: https://vault.centos.org/centos/7.9.2009/os/Source/SPackages/perl-HTTP-Daemon-6.01-8.el7.src.rpm

rzr replied:
Packages are now imported, we can run tests, doing this now:

- [x] Koji pre-build XCPNG2710 (v8.3-u-pcoval2): https://koji.xcp-ng.org/taskinfo?taskID=97280

rzr commented:
> upstream is too different from XS, and fedora f39 did have a 1.21

I note that fedora switched to wget2 just after 1.21.3

https://src.fedoraproject.org/rpms/wget.git

And upstream released a later version:

https://ftp.gnu.org/gnu/wget/wget-1.25.0.tar.gz

Will keep you up to date what is the most meaningful.

rzr committed:
Make check optional for xcp-ng-8.3 (enabled by default)

This can be useful if deps are not available.

The dependency to "perl-HTTP-Daemon-0:6.01-8.el7.noarch to provide
'perl(HTTP::Daemon)' brings more dependencies:

- perl-HTTP-Daemon
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/perl-HTTP-Daemon-6.01-8.el7.src.rpm
- perl-HTTP-Date
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/perl-HTTP-Date-6.02-8.el7.src.rpm
- perl-HTTP-Message
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/perl-HTTP-Message-6.06-6.el7.src.rpm
- perl-IO-Socket-IP
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/perl-IO-Socket-IP-0.21-5.el7.src.rpm
- perl-LWP-MediaTypes
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/perl-LWP-MediaTypes-6.02-2.el7.src.rpm
- mailcap
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/mailcap-2.1.41-2.el7.src.rpm
- perl-Business-ISBN
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/perl-Business-ISBN-2.06-2.el7.src.rpm
- perl-Business-ISBN-Data
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/perl-Business-ISBN-Data-20120719.001-2.el7.src.rpm
- perl-IO-HTML
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/perl-IO-HTML-1.00-2.el7.src.rpm
- perl-URI
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/perl-URI-1.60-9.el7.src.rpm
- perl-Encode-Locale
  - https://vault.centos.org/7.9.2009/os/Source/SPackages/perl-Encode-Locale-1.03-5.el7.src.rpm

- perl-Compress-Raw-Bzip2
  - https://updates.xcp-ng.org/8/8.3/base/Source/SPackages/perl-Compress-Raw-Bzip2-2.061-3.el7.src.rpm
- perl-Compress-Raw-Zlib
  - https://updates.xcp-ng.org/8/8.3/base/Source/SPackages/perl-Compress-Raw-Zlib-2.061-4.el7.src.rpm
- perl-IO-Compress
  - https://updates.xcp-ng.org/8/8.3/base/Source/SPackages/perl-IO-Compress-2.061-2.el7.src.rpm
- perl-TimeDate
  - https://updates.xcp-ng.org/8/8.3/base/Source/SPackages/perl-TimeDate-2.30-2.el7.src.rpm

Skiping this test, can avoid to import (and maintain) more package, at
the price of less testing at build time.

This change has been inspired from change in wget-1.21.3-3.xs8.src.rpm

Origin: https://github.com/xcp-ng-rpms/wget/pull/1
Signed-off-by: Philippe Coval <philippe.coval@vates.tech>

rzr committed:
Import wget-1.21.3-3.fc37.src.rpm

Signed-off-by: Philippe Coval <philippe.coval@vates.tech>
Origin: https://kojipkgs.fedoraproject.org//packages/wget/1.21.3/3.fc37/src/wget-1.21.3-3.fc37.src.rpm

rzr committed:
Import wget-1.21.4-1.fc39.src.rpm

Signed-off-by: Philippe Coval <philippe.coval@vates.tech>

rzr commented:
Please add perl module `perl-HTTP-Daemon` to koji incoming branch:


- https://koji.xcp-ng.org/kojifiles/work/tasks/8115/98115/root.log


```
BuildRequires: perl(HTTP::Daemon)
```

```
DEBUG util.py:461:  Error: No Package found for perl(HTTP::Daemon)
```

rzr commented:
- [x] Koji build (v8.3-incoming): https://koji.xcp-ng.org/taskinfo?taskID=98124#wget#fbaeb96
